### PR TITLE
docs: document more features

### PR DESCRIPTION
**Pull request by g-linville**
- I moved the info about workspaces from the FAQ page to its own page.
- I added a page to provide additional guidelines about how to structure code-based tools in GitHub repos.
- I added a page about daemon tools.
- I added a section to the FAQ about saving and resuming chat state.

### CODE REVIEW

File: docs/docs/03-tools/10-daemon.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+Let's talk about the daemon tool, called `my-daemon`, first.
+
+### The Daemon Tool
+
+The body of this tool begins with `#!sys.daemon`. This tells GPTScript to take the rest of the body as a command to be
+run in the background that will listen for HTTP requests. GPTScript will run this command (in this case, a Node script).
+GPTScript will assign a port number for the server and set the `PORT` environment variable to that number, so the
+server needs to check that variable and listen on the proper port. The server is required to respond to a GET request at
+`/` with 200 OK, so that GPTScript can make sure it is running properly before continuing execution.
```
Comment (thedadams) [inline docs/docs/03-tools/10-daemon.md:62]: The path here is actually configurable. Do we want to mention that?

See the [UI](https://github.com/gptscript-ai/ui/blob/main/tool.gpt) for an example with the `(path ...)` option.

File: docs/docs/03-tools/10-daemon.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+```
+
+Let's talk about the daemon tool, called `my-daemon`, first.
+
+### The Daemon Tool
+
+The body of this tool begins with `#!sys.daemon`. This tells GPTScript to take the rest of the body as a command to be
+run in the background that will listen for HTTP requests. GPTScript will run this command (in this case, a Node script).
+GPTScript will assign a port number for the server and set the `PORT` environment variable to that number, so the
+server needs to check that variable and listen on the proper port. The server is required to respond to a GET request at
```
[the remark this answers is not in the document]
Comment (g-linville) [reply docs/docs/03-tools/10-daemon.md:61]: I'm a bit confused by this. In general, the daemon will receive POSTs from other tools. But GPTScript itself just sends a GET request to make sure it is alive.

File: docs/docs/03-tools/10-daemon.md (hunk 1 above)
Comment (g-linville) [reply docs/docs/03-tools/10-daemon.md:62]: Thanks for mentioning this - I wasn't aware this was a thing. I'll add it to the doc

File: docs/docs/03-tools/08-workspace.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Regardless of whether it is set implicitly or explicitly, the workspace is then made available to the script execution as the `GPTSCRIPT_WORKSPACE_DIR` environment variable.
+
+:::info
+GPTScript does not force scripts or tools to write to, read from, or otherwise use the workspace.
+The tools must decide to make use of the workspace environment variable.
+:::
+
+## The Workspace Context Tool
+
+To make prompt-based tools aware of the workspace, you can reference the workspace context tool:
```
[the remark this answers is not in the document]
Comment (g-linville) [reply docs/docs/03-tools/08-workspace.md:24]: Yep, I added more clarification to this

**Review by thedadams — APPROVED** [review]

**Review by iwilltry42 — APPROVED** [review]
Very nice! Learned something 😁

**Pull request merged**